### PR TITLE
fix #103. update dependencies

### LINKED ISSUE #103: update dependencies
Ideally update all dependencies to whatever the current version is, especially RxJS, but certainly whatever the latest version that can be used without major code changes.

### PR DESCRIPTION
**Pull request by james-anderson**
Related to #3 update for React 16 and 17
In testing a project using React 16.4 and 17.0, I found that I had to update Babel dependencies and `.babelrc` files in those projects to get the project to compile with the `conduit` updates. It is possible it was just the way I went about testing it or that the React 16.4 project had other issues with its dependencies. If it's not just me and the projects I tested with, we may want to include the Babel dependencies in `peerDependencies` to produce a warning on `npm install` rather than compile errors when building the project. We would probably want to update the `peerDependencies` for React (included in this PR) due to the function name change of `componentWillReceiveProps` to `UNSAFE_componentWillReceiveProps` in `conduit-rxjs-react`. I considered including both versions of the function name thinking older versions of React would notice the old name and ignore the new one and newer versions would ignore the old one and use the new one, but React was throwing warnings about the old name since it will still be recognized until v18, so I just replaced it. It seems like the RxJS `peerDependencies` can stay as-is.

### CODE REVIEW

File: packages/conduit-rxjs-react/package.json
Comment (basham) [inline packages/conduit-rxjs-react/package.json:35]: Since we know there's a top-range for compatibility, we can add that in here. I didn't find anything reliable that explicitly states it, but I'm imagining that `UNSAFE_componentWillReceiveProps` will be removed in v18?

```
"react": ">= 16.3.0 <18",
```

File: packages/conduit-rxjs-react/CHANGELOG.md
Comment (basham) [inline packages/conduit-rxjs-react/CHANGELOG.md:9]: Are we sure that it is "to 18.x" and not "to 17.x"? I couldn't find a source one way or another. But it seems odd that the UNSAFE methods would carry over to a second major version.

File: examples/todomvc/webpack.config.js
Comment (basham) [inline examples/todomvc/webpack.config.js:17]: As of [RxJS v7.2](https://github.com/ReactiveX/rxjs/blob/master/CHANGELOG.md#720-2021-07-05) (2021-07-05):

> Operators are all exported at the top level, from "rxjs". From here on out, we encourage top-level imports with RxJS. Importing from rxjs/operators will be deprecated soon.

So, if we upgrade the examples to the latest RxJS, then we can clean up this "externals" line — here and in all the examples. We could also force the minimum RxJS version to be v7.2, if we wanted. It would primarily affect `conduit-rxjs-react` since it uses operators, while `conduit-rxjs` does not. But if we force a minimum version on one, we may as well do the same for the other.

**Review by basham — CHANGES_REQUESTED** [review]
This is really fantastic work! Do we want to also remove the `package-lock.json` files in this PR? It seems like it is in scope.

File: packages/conduit-rxjs-react/package.json
Comment (james-anderson) [reply packages/conduit-rxjs-react/package.json:35]: All the information I had previously looked at made me think `componentWillReceiveProps` will be removed in React v17, but when testing updates to SER Cody discovered that it still existed. He found a blog post https://reactjs.org/blog/2020/10/20/react-v17.html that makes it sound like the breaking changes expected in v17 were postponed until v18. I expect v18 will now remove `componentWillReceiveProps` and throw warnings about `UNSAFE_componentWillReceiveProps`. This seems to follow from the React v17 code. There it has an automatic warning for using `componentWillReceiveProps`, but only warns about `UNSAFE_componentWillReceiveProps` in "strict mode". I expect in v18 `componentWillReceiveProps` will be cause an error and `UNSAFE_componentWillReceiveProps` will cause a warning whether you have "strict mode" enabled or not.

We could assume for now that we will not be compatible with v18 and update to say that we are if we find out that `UNSAFE_componentWillReceiveProps` will still work in v18.

File: examples/todomvc/webpack.config.js
Comment (james-anderson) [reply examples/todomvc/webpack.config.js:17]: 👍 Since we're forcing React updates we might as well also force RxJS updates.

File: packages/conduit-rxjs-react/package.json
Comment (basham) [reply packages/conduit-rxjs-react/package.json:35]: Investigating a little. [According to the blog](https://reactjs.org/blog/2021/06/08/the-plan-for-react-18.html#how-to-try-react-18-alpha-today):

> New alphas are regularly published to npm using the @alpha tag. These releases are built using the most recent commit to our main repo.

Searching the `main` branch (which should be the latest stuff going into v18), there are [some unit tests for `componentWillReceiveProps`](https://github.com/facebook/react/blob/cae635054e17a6f107a39d328649137b83f25972/packages/react/src/__tests__/createReactClassIntegration-test.js). These tests were last updated in August 2020. These tests still suggest that React invokes the legacy lifecycle methods under certain conditions. These tests don't seem to fully align yet with what I'd expect, given the stated depreciations.

Given this uncertainty, I revoke this suggestion. I don't want to put in the upper limit (`<18`) anymore. This particular Conduit version may break in some future version of React, but that's the general expectation for any software dependencies, anyway. And we will make an update soon enough to make this a null issue.

In summary — never mind.

File: examples/todomvc/webpack.config.js
Comment (james-anderson) [reply examples/todomvc/webpack.config.js:17]: Would it be appropriate to replace the references to rxjs/operators in the CHANGELOG.md files too?

Comment (basham) [reply examples/todomvc/webpack.config.js:17]: The README files and examples should be updated. But operator imports mentioned in prior releases in the CHANGELOG files should be kept as-is. It should reflect the code at that point in time. This change should be mentioned in the CHANGELOG for this upcoming version.

File: packages/conduit-rxjs-react/package.json
Comment (james-anderson) [reply packages/conduit-rxjs-react/package.json]: I didn't notice this message until after I put in the change. I'll remove the "< 18".

File: examples/todomvc/webpack.config.js
Comment (basham) [reply examples/todomvc/webpack.config.js]: Since all operators are now a top-level import, I'd have one import line, rather than two.

```js
// Update this:
import { interval } from 'rxjs'
import { map, startWith, switchMap } from 'rxjs'

// To this (organizing imports alphabetically):
import { interval, map, startWith, switchMap } from 'rxjs'
```

File: packages/conduit-rxjs-react/CHANGELOG.md
Comment (basham) [inline packages/conduit-rxjs-react/CHANGELOG.md:11]: This could be updated with a direct link.

```
3. Switched `rxjs/operators` imports to `rxjs` since they moved to top level in [RxJS v7.2](https://github.com/ReactiveX/rxjs/blob/master/CHANGELOG.md#720-2021-07-05).
```

**Review by basham — CHANGES_REQUESTED** [review]
Also:

1. Merge multiple `rxjs` imports to one import in all MD files. (Thanks for doing it in the JS files.)
2. Remove `package-lock.json` files.

File: packages/conduit-rxjs-react/CHANGELOG.md
Comment (basham) [inline packages/conduit-rxjs-react/CHANGELOG.md:11]: I want to be more explicit about the new peer dependency version. Also, make this item 2, rather than item 3 in this log. That way, this `conduit-rxjs-react` log will start identically to the `conduit-rxjs` log.

```md
2. Increased minimum peer dependency version of [RxJS to v7.2](https://github.com/ReactiveX/rxjs/blob/master/CHANGELOG.md#720-2021-07-05). Operators are now imported from the top-level `rxjs` module, since importing from `rxjs/operators` will be depreciated soon.
```

File: packages/conduit-rxjs/CHANGELOG.md
Comment (basham) [inline packages/conduit-rxjs/CHANGELOG.md:7]: Add this item. Even though `conduit-rxjs` doesn't use any operators, I'm fine leaving in the second sentence, so that there's an explicit reason for the change.

```md
2. Increased minimum peer dependency version of [RxJS to v7.2](https://github.com/ReactiveX/rxjs/blob/master/CHANGELOG.md#720-2021-07-05). Operators are now imported from the top-level `rxjs` module, since importing from `rxjs/operators` will be depreciated soon.
```

**Review by basham — CHANGES_REQUESTED** [review]
Just a few minor changes to the logs.

**Review by basham — CHANGES_REQUESTED** [review]
Also, the examples are broken. The `index.html` files need updated versions. I'd recommend React be v16.3.0, since that's the minimum version we're asking for. And likewise, RxJS be v7.2.0. Also, note that the bundle folder for RxJS has changed. It's now under `dist`.

```html
  <script crossorigin src="https://unpkg.com/react@16.3.0/umd/react.production.min.js"></script>
  <script crossorigin src="https://unpkg.com/react-dom@16.3.0/umd/react-dom.production.min.js"></script>
  <script crossorigin src="https://unpkg.com/rxjs@7.2.0/dist/bundles/rxjs.umd.min.js"></script>
```

**Comment by basham** [comment]
Once this PR merges, perhaps we do one minor PR to update the publish dates in the change logs. Then we publish. These are all plenty of updates for this next release.

**Comment by james-anderson** [comment]
@basham, I've got the most recent requested changes done. I'm good with doing a minor PR to set the publish dates just before publishing.

**Review by basham — APPROVED** [review]

**Comment by basham** [comment]
@james-anderson The examples all work. It looks good to me. Thanks for all your work on this! I'll leave it to you to merge, whenever you're ready to do so. Do you want to work on the next PR and publish the changes to npm? You should still have the ability to do the publishing, if you want to do that. If not, I can take care of it. Up to you.

**Comment by james-anderson** [comment]
@basham I can do the publish date PR tomorrow and publish following the CONTRIBUTING.md steps.

**Pull request merged**
